File: js/javascript-exercises/pig_latin/pigLatin.js
function translate(string) {
	const words = string.split(' ');
	var index;
	const coded = words.map(word => {
	  
	  const vowels = word.match(/[aeiou]/g);
	  if( vowels[0]=='u' && word[word.indexOf(vowels[0])-1]=='q') {
	    index = word.indexOf(vowels[0])+1;
	  } else { 
	    index = word.indexOf(vowels[0]); 
	    }
	 
	  return word.slice(index) + word.substr(0,index) + "ay";
	  
  })
  
	return coded.join(' ');
	
	}
	


module.exports = {
	translate }

File: ruby/learn_ruby/04_pig_latin/pig_latin.rb
def translate(phrase)
  pigLatin = phrase.split(' ').map do |word|
    if word[0].index(/[aeiou]/)
      pigLatin = "#{word}ay"
    else
      indexOfU = word.index('u')
      if indexOfU && word[indexOfU-1] == 'q'
        pigLatin = word[indexOfU+1..-1]+word[0..indexOfU] +"ay"
      else
        index = word.index(/[aeiou]/)
        pigLatin = word[index..-1]+word[0..index-1] +"ay"
      end
    end
  end
  pigLatin.join(' ')
end
